File: Infrastructure/Logic/IAuthorDM.cs
using System;
using System.Collections.Generic;
using ViewModels;

namespace Infrastructure.Logic
{
    public interface IAuthorDM : IDisposable
    {
        IEnumerable<AuthorVM> Get(string term);
        DataTableResponseVM<AuthorVM> GetList(DataTableRequestVM model);
        AuthorVM Get(long? id);
        void Delete(long id);
        void Update(AuthorVM model);
        void Create(AuthorVM model);
    }
}

File: InfrastructureVB/Logic/IAuthorDM.vb
Imports ViewModels

Namespace Logic
    Public Interface IAuthorDM
        Inherits IDisposable

        Sub Update(model As AuthorVM)
        Sub Create(model As AuthorVM)
        Function [Get](model As Long?) As AuthorVM
        Sub Delete(id As Long)

        Function GetList(model As DataTableRequestVM) As DataTableResponseVM(Of AuthorVM)
        Function [Get](term As String) As IEnumerable(Of AuthorVM)

    End Interface
End Namespace
